### DISCUSSION TITLE
Named Timeseries with percentile()

### DISCUSSION THREAD

**Discussion by tobiaszuercher** [ravendb/ravendb#14251] [Q&A]
Hi

given a LapTime (simplified)
   * Laptime (in ms)
   * PlayerRating (0-15000)

i want to create some statistics for the laptimes.

first try: a map reduce index, where in the reduce i do create buckets for the Player Ratings (e.g. `(int)(Math.Floor((double)lap.PlayerRating/ 1000) * 1000)` to get 1k, 2k, 3k... buckets).

I also use standard Max/Min/Avg for the Laptime itself.

Is there a chance to do percentile instead of Min/Max? We want to filter out the extremes...

As far as i understand this is not possible with a map reduce index. I'm happy to be wrong :)

Next try: Timeseries with percentile() (https://ravendb.net/docs/article-page/5.3/csharp/document-extensions/timeseries/querying/statistics)

We do use Named Timeseries & i was wondering how it works in c# with Named Timeseries, since i can't find it in the docs.

I tried in the studio:

```from 'xxxx'
where id() == 'xxxxx'
select timeseries(from 'Laps'
select percentile(90))
```

i do get this response:
<img width="229" alt="image" src="https://user-images.githubusercontent.com/516373/169695035-d64e0792-9865-4021-9c0f-9458044f206d.png">
Seems there is the percentile of every field.

is there a way to get the percentile only for a specific field?

In words: percentile(90) for laptime and then after min/max/avg.

**Comment by georgiosd** [comment]
@tobiaszuercher I've used this successfully in a map-reduce index!

https://stackoverflow.com/questions/8137391/percentile-calculation/8137526#8137526

**Comment by ayende** [comment]
There isn't an easy way to get just the data for a specific field, no. Easiest is just to query them and do the processing on your end.
